pkg/gcp/gcp.go

```diff
@@ -197,3 +240,27 @@ func (c *GCSClient) Close(ctx context.Context) {
 func (c *GCSClient) ObjectIsNotFound(err error) bool {
 	return errors.Is(err, gcpstorage.ErrObjectNotExist)
 }
+
+// newHTTPClient creates a new HTTP client for interacting with Google Cloud APIs.
+func newHTTPClient(ctx context.Context, o *options) (*http.Client, error) {
+	baseTransport := http.DefaultTransport.(*http.Transport).Clone()
+	if o.proxyURL != nil {
+		baseTransport.Proxy = http.ProxyURL(o.proxyURL)
+	}
+
+	var opts []option.ClientOption
+
+	if o.secret != nil {
+		creds, err := google.CredentialsFromJSON(ctx, o.secret.Data["serviceaccount"], gcpstorage.ScopeReadOnly)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create Google credentials from secret: %w", err)
+		}
+		opts = append(opts, option.WithCredentials(creds))
+	}
+
+	transport, err := htransport.NewTransport(ctx, baseTransport, opts...)
```

Comment:
Good question, there's actually a good reason (I will add a comment about that in the code).

The reason is: If we just pass `option.WithCredentialsJSON()` to `htransport.NewTransport()`, we get the following error:

```
Get "https://storage.googleapis.com/storage/v1/b/<bucket-name>?alt=json&prettyPrint=false&projection=full": oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"invalid_scope","error_description":"Invalid OAuth scope or ID token audience provided."}
```

Because the function `gcpstorage.NewClient()` is specific to GCS, it adds the `gcpstorage.ScopeReadOnly` scope automatically. Since we need `htransport.NewTransport()` in this case, we must use `google.CredentialsFromJSON()`, which allows specifying scopes. That function returns a credential that must be passed via `option.WithCredentials()`.

Reply:
Commend added, thanks!